pkg/pbq/pbq_test.go

```diff
@@ -0,0 +1,202 @@
+	pq, err := NewPBQ(ctx, "partition-1", memStore, qManager, options)
+	assert.NoError(t, err)
+
+	for _, msg := range writeMessages {
+		err := pq.WriteFromISB(&msg)
+		assert.NoError(t, err)
+	}
+
+	// check if the messages are persisted in store
+	storeMessages, _ := pq.Store.ReadFromStore(10)
+	assert.Len(t, storeMessages, msgCount)
+	pq.CloseOfBook()
+	// this means we successfully wrote 10 messages to pbq
+}
+
+func TestPBQ_ReadFromPBQ(t *testing.T) {
+	// create a store of size 100 (it can store max 100 messages)
+	storeSize := 100
+	options := &store.Options{}
+	_ = store.WithPbqStoreType("in-memory")(options)
+	_ = store.WithStoreSize(int64(storeSize))(options)
+	_ = store.WithReadTimeout(1)(options)
+
+	ctx := context.Background()
+
+	qManager, _ := NewManager(ctx, store.WithPbqStoreType("in-memory"), store.WithStoreSize(int64(storeSize)))
+	memStore, err := memory.NewMemoryStore(ctx, "partition-1", options)
+	assert.NoError(t, err)
+
+	//write 10 isb messages to persisted store
+	msgCount := 10
+	startTime := time.Now()
+	writeMessages := testutils.BuildTestWriteMessages(int64(msgCount), startTime)
+
+	// lets create a pbq with buffer size 10
+	buffSize := 10
+	_ = store.WithBufferSize(int64(buffSize))(options)
+	pq, err := NewPBQ(ctx, "new-partition", memStore, qManager, options)
+	assert.NoError(t, err)
+
+	for _, msg := range writeMessages {
+		err := pq.WriteFromISB(&msg)
+		assert.NoError(t, err)
+	}
+
+	pq.CloseOfBook()
+	readMessages, _ := pq.ReadFromPBQ(ctx, int64(100))
+	// number of messages written should be equal to number of messages read
+	assert.Len(t, readMessages, msgCount)
+	err = pq.GC()
+	assert.NoError(t, err)
+}
+
+func TestPBQ_ReadWrite(t *testing.T) {
+	// create a store of size 100 (it can store max 100 messages)
+	storeSize := 100
+	options := &store.Options{}
+	_ = store.WithPbqStoreType("in-memory")(options)
+	_ = store.WithStoreSize(int64(storeSize))(options)
+	_ = store.WithReadTimeout(1)(options)
+
+	ctx := context.Background()
+
+	qManager, _ := NewManager(ctx, store.WithPbqStoreType("in-memory"), store.WithStoreSize(int64(storeSize)))
+	memStore, err := memory.NewMemoryStore(ctx, "partition-1", options)
+	assert.NoError(t, err)
+
+	//write 10 isb messages to persisted store
+	msgCount := 10
+	startTime := time.Now()
+	writeMessages := testutils.BuildTestWriteMessages(int64(msgCount), startTime)
+
+	//create a pbq with buffer size 10
+	buffSize := 10
+	_ = store.WithBufferSize(int64(buffSize))(options)
+	pq, err := NewPBQ(ctx, "new-partition", memStore, qManager, options)
+	assert.NoError(t, err)
+
+	var readMessages []*isb.Message
+	// run a parallel go routine which reads from pbq and writes to a channel
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		for {
+			msgs, err := pq.ReadFromPBQ(context.Background(), 1)
+			readMessages = append(readMessages, msgs...)
+			if err != nil {
+				break
+			}
+		}
+		err := pq.Close()
+		assert.NoError(t, err)
```

Comment:
can you add a test for `EOF`, since `cob` has happened?

Reply:
done